### ISSUE TITLE
Upgrade to Reek 2.0.3 which fixes calling super with parameters

### ISSUE THREAD

**Issue by eostrom** [whitesmith/rubycritic#45]
RubyCritic started crashing when run on my codebase:

```
/home/vagrant/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/reek-1.6.5/lib/reek/smells/nested_iterators.rb:73:in `ignored_iterator?': undefined method `method_name' for (super):#<Class:0x007f189d1c0180> (NoMethodError)
```

Reek 2.0 has what [looks like a fix](https://github.com/troessner/reek/commit/7b9af8c63c3f2ca40d47e378a12398122d46706d): "Make NestedIterator not break when iterator is called on super."

I forked the RubyCritic code and tried updating Reek. The tests now fail for two reasons:

  * `samples/reek/smelly.rb` now has two smells, not one; the new one is IrresponsibleModule.
  * `samples/reek/not_smelly.rb` has a lot of smells, not none.

I think the issue in both cases is that `config.reek` is being ignored. The Reek changelog lists a couple of possibly related items involving the Rake task and CLI options.

I'm going to look into that and hopefully submit a PR, but I thought I would check in with what I have so far.

**Comment by LeeXGreen** [comment]
I think this fix *should* be in Reek 1.6.5 as well (see https://github.com/troessner/reek/issues/379), so I'm not sure why you're seeing this.

**Comment by eostrom** [comment]
Good catch, thanks. I guess I have to actually debug.

**Comment by eostrom** [comment]
It looks like the fix handles `super` but not `super()`, which is necessary if, say, you want to make an `Enumerator` subclass whose initializer takes parameters. I'll pursue that over in the Reek repo.

**Comment by LeeXGreen** [comment]
Thanks for the debug :)

**Comment by topher6345** [comment]
I'm also getting this error from rubycritic via reek 1.6.3
`~/.rvm/gems/ruby-2.1.5/gems/reek-1.6.3/lib/reek/smells/nested_iterators.rb:73:in `ignored_iterator?': undefined method `method_name' for (zsuper):#<Class:0x007f9d15c82418> (NoMethodError)`

the code that triggers the error is this: 

```ruby
....
  def initialize(params)
    super do
      @model = MyClass.create! myclass_params
    end
  end
....
```

**Comment by eostrom** [comment]
That one should be fixed in the current release of RubyCritic, which uses
Reek 1.6.5.

**Comment by topher6345** [comment]
Works beautifully in RubyCritic 1.4.0. I wasn't paying attention to my Gemfile. Sorry for the false bug. Thanks a bunch for the timely response.

**Comment by pedrobachiega** [comment]
Still happening here:
```
/Users/pedrobachiega/.rvm/gems/ruby-2.1.5/gems/reek-1.6.5/lib/reek/smells/nested_iterators.rb:73:in `ignored_iterator?': undefined method `method_name' for (super (NoMethodError)
  (lvar :options)):#<Class:0x007f9a51a6abb0>
```
```
ruby-2.1.5
rubycritic-1.4.0
reek-1.6.5
```

some help, please?

**Comment by eostrom** [comment]
@pedrobachiega If your code specifies parameters to `super` (i.e., you call `super()` instead of `super`), this is probably a [Reek bug](https://github.com/troessner/reek/issues/404). A [fix](https://github.com/troessner/reek/pull/424) has been [released in Reek 2.0.4](https://github.com/troessner/reek/compare/v2.0.2...v2.0.3), and [merged into RubyCritic master](https://github.com/whitesmith/rubycritic/pull/49), so it should be in the next release.

In the meantime, try [specifying RubyCritic master in your Gemfile](http://bundler.io/git.html):

```ruby
gem 'rubycritic', github: 'whitesmith/rubycritic'
```

If that doesn't help, it's still probably a Reek issue; try reporting it there, and include a test case to help them debug it.

**Comment by pedrobachiega** [comment]
thanks @eostrom !
using the master from github worked!

**Comment by troessner** [comment]
We're now on Reek 3.6, closing this one.

**Comment by LeeXGreen** [comment]
Version 2.1 is on Reek 3.6, which is not compatible with Ruby 1.9.

Do you plan to release the bugfix commits on master, after the v1.4.0 tag?

Are there any plans to release bugfixes for the v1.X series of RubyCritic?

**Comment by natesholland** [comment]
And a follow up on that: can we make a branch for 1.4 so that if we want to make pull requests for bug fixes for 1.4 we have a good target to point it at?

**Comment by Onumis** [comment]
We should focus on the current ruby versions and not back-port fixes as that generally is too much effort for little proffit.
That being said, I see no problem on doing eventual fixes to v1.4, but never new features.

**Comment by troessner** [comment]
>> We should focus on the current ruby versions and not back-port fixes as that generally is too much effort for little proffit.
That being said, I see no problem on doing eventual fixes to v1.4, but never new features.

Couldn't agree more.
